Resolver/RemotePageResolver.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mooore\WordpressIntegrationCms\Resolver;
+
+use Mooore\WordpressIntegrationCms\Model\RemotePageRepository;
+
+class RemotePageResolver
+{
+    /**
+     * @var RemotePageRepository
+     */
+    private $pageRepository;
+    /**
+     * @var array
+     */
+    private $remotePageContentCache = [];
+
+    public function __construct(RemotePageRepository $pageRepository)
+    {
+        $this->pageRepository = $pageRepository;
+    }
+
+    public function resolve(int $siteId, int $pageId)
+    {
+        $cacheKey = sprintf('page_%s_%s', $siteId, $pageId);
+
+        if (isset($this->remotePageContentCache[$cacheKey])) {
+            return $this->remotePageContentCache[$cacheKey];
+        }
+
+        try {
+            $remotePage = $this->pageRepository->get((int) $siteId, (int) $pageId);
```

Comment:
Isn't $siteId, and $pageId always int, as it's set in the arguments of the functions, anyways?